File: code/Cpp/Strings/Day_17_FirstNonRepChar.cpp
class Solution {
  public:

    char nonRepeatingChar(string &s) {

        unordered_map<char,int> d;
        
        for(char c:s){
            d[c]++;
        }
        
        for(char c:s){
            if(d[c]==1) return c;
        }
        return '$';

    }
};

File: code/Python/Strings/Day_17_FirstNonRepChar.py
class Solution:
    
    #Function to find the first non-repeating character in a string.
    def nonRepeatingChar(self,s):
        
        d={}
        for i in s:
            d[i] = d.get(i,0)+1
        
        for i in s:
            if d[i]==1:
                return i
        
        return "$"